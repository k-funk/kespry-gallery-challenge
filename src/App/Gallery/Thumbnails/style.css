.thumbnail {
  --thumbnail-size: 50px;

  cursor: pointer;
  opacity: 1;
  padding: 2px;
  height: var(--thumbnail-size);
  transition: opacity var(--transition-time);
  width: var(--thumbnail-size);
}
.thumbnail:hover {
  opacity: .8;
}

.thumbnail::selection {
  background: transparent;
}