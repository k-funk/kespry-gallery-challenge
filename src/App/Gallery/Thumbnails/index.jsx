import React from 'react';
import { PropTypes as T } from 'prop-types';

import './style.css';


const Thumbnails = ({ imageList, selectImage }) => (
  <div className="thumbnails">
    {imageList.map((imageUrl, idx) => (
      <span
        key={imageUrl}
        role="button"
        tabIndex={0}
        onClick={() => selectImage(imageUrl)}
        onKeyUp={() => selectImage(imageUrl)}
      >
        <img
          src={imageUrl}
          alt={`Thumbnail ${idx}`}
          className="thumbnail"
        />
      </span>
    ))}
  </div>
);

Thumbnails.propTypes = {
  imageList: T.arrayOf(T.string).isRequired,
  selectImage: T.func.isRequired,
};

export default Thumbnails;
