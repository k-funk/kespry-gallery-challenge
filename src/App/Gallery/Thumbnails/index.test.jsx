import React from 'react';
import { shallow } from 'enzyme';
import shallowToJson from 'enzyme-to-json';

import Thumbnails from './index';


describe('outputs the expected tree when', () => {
  let wrapper;
  let selectImageSpy;

  beforeEach(() => {
    selectImageSpy = jest.fn();
  });

  test('(default)', () => {
    wrapper = shallow((
      <Thumbnails
        imageList={['https://img.com/1', 'https://img.com/2']}
        selectImage={selectImageSpy}
      />
    ));
  });

  afterEach(() => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
