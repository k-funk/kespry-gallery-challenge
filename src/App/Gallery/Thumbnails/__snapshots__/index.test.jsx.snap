// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`outputs the expected tree when (default) 1`] = `
<div
  className="thumbnails"
>
  <span
    key="https://img.com/1"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <img
      alt="Thumbnail 0"
      className="thumbnail"
      src="https://img.com/1"
    />
  </span>
  <span
    key="https://img.com/2"
    onClick={[Function]}
    onKeyUp={[Function]}
    role="button"
    tabIndex={0}
  >
    <img
      alt="Thumbnail 1"
      className="thumbnail"
      src="https://img.com/2"
    />
  </span>
</div>
`;
